### PR TITLE
Move task types and categories to their own TaskType and TaskCategory .proto enums

### PR DESCRIPTION
**Pull request by markmark206**
(apologies for a gazillion of commits – this is due to managing a sequence of cascading changes, going through PRs in a sequence –  they will be squished when this PR merges)

**What changed?**
Moving task types into their own TaskType `.proto` `enum`.
Moving task categories into their own TaskCategory `.proto` `enum`.
Remove unused `NamespaceConfig.archivalStatus`.

**Why?**
This is part of our general effort to clean up the code, and make it stronger typed and more consistent.

**How did you test it?**
I compiled the code and ran unit tests.

**Potential risks**
This is purely code refactoring / clean up and should cause no behavior changes. If anything breaks, tests should be able to catch that, and we will fix or revert.

### CODE REVIEW

File: proto/adminservice/server_enum.proto [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+syntax = "proto3";
+
+package adminservice;
+
+option go_package = "github.com/temporalio/temporal/.gen/proto/adminservice";
+
+enum TaskCategory {
```
Comment (alexshtin) [inline proto/adminservice/server_enum.proto:27]: Move this to `common/server_enum.proto`. And remove this file.

File: proto/historyservice/request_response.proto [hunk 2]
```diff
@@ -84,7 +86,7 @@ message GetMutableStateResponse {
     // TODO: when migrating to gRPC, make this a enum
     // TODO: when migrating to gRPC, unify internal & external representation
     // NOTE: workflowState & workflowStatus are the same as persistence representation (???)
-    int32 workflowState = 16;
+    execution.WorkflowExecutionState workflowState = 16;
```
Comment (alexshtin) [inline proto/historyservice/request_response.proto:89]: I think you can remove all comments above now.

File: go.sum [hunk 3]
```diff
@@ -95,6 +95,8 @@ github.com/golang/protobuf v1.4.0-rc.2/go.mod h1:LlEzMj4AhA7rCAGe4KMBDvJI+AwstrU
 github.com/golang/protobuf v1.4.0-rc.4.0.20200313231945-b860323f09d0/go.mod h1:WU3c8KckQ9AFe+yFwt9sWVRKCVIyN9cPHBJSNnbL67w=
 github.com/golang/protobuf v1.4.0 h1:oOuy+ugB+P/kBdUnG5QaMXSIyJ1q38wWSojYCb3z5VQ=
 github.com/golang/protobuf v1.4.0/go.mod h1:jodUvKwWbYaEsadDk5Fwe5c77LiNKVO9IDvqG2KuDX0=
+github.com/golang/protobuf v1.4.1 h1:ZFgWrT+bLgsYPirOnRfKLYJLvssAegOj/hgyMFdJZe0=
```
Comment (alexshtin) [inline go.sum:98]: Note: if you see line changed in `go.mod` but new lines added in `go.sum` it means you need to run `go mod tidy`.

Comment (markmark206) [reply go.sum:98]: Done!

File: proto/historyservice/request_response.proto (hunk 2 above)
Comment (markmark206) [reply proto/historyservice/request_response.proto:85]: Done!

**Pull request merged**